src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TypeMetadataNode.cs

```diff
@@ -58,6 +58,15 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
                 // A lot of the enum reflection actually happens on top of the respective MethodTable (e.g. getting the underlying type),
                 // so for enums also include their MethodTable.
                 dependencies.Add(factory.MaximallyConstructableType(_type), "Reflectable enum");
+
+                // Enums are not useful without their literal fields
```

Comment:
I understand the comment, but my question is why do we need to explicitly preserve the metadata for the fields - I would expect that if the app uses them we would see their usage directly. Or is this because annotations/suppressions in framework assume this? (If so I think it's worth a comment about that).

Reply:
The fields are never referenced from code. Basically, an enum looks like this:

```csharp
class MyEnum : Enum
{
    private int _value;
    public const int Value1 = 1;
    public const int Value2 = 2;
}
```

One cannot do `ldsfld MyEnum.Value1` in IL. It's lowered into `ldc.i4.1`.

Added comment.